include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,346 @@
+      storage.size_     = 0;
+    }
+
+    //! @brief Removes an element from the container.
+    //! Has the same invalidation semantics as std::vector.
+    //! end() iterator is not a valid pos.
+    // iterator erase(const_iterator pos)
+    // {
+    //   std::ptrdiff_t distance = pos - cbegin();
+    //   kumi::for_each([&](auto& m) { return m.erase(m.begin() + distance); }, storage);
+    //   return begin() + distance;
+    // }
+
+    //! @brief Removes the elements in the range [first, last)
+    //! Empty range is OK, does nothing
+    // iterator erase(const_iterator f, const_iterator l)
+    // {
+    //   std::ptrdiff_t distance_f = f - cbegin();
+    //   std::ptrdiff_t distance_l = l - cbegin();
+    //   kumi::for_each([&](auto& m) {
+    //     return m.erase(m.begin() + distance_f, m.begin() + distance_l); }, storage);
+    //   return begin() + distance_f;
+    // }
+
+    //! @brief Appends the given element value to the end of the container.
+    //! If the new size() is greater than capacity() then all iterators and references (including
+    //! the past-the-end iterator) are invalidated. Otherwise only the past-the-end iterator is
+    //! invalidated.
+    //! @param value [Value](@eve::vectorizable) to append
+    // EVE_FORCEINLINE void push_back(value_type const& value) noexcept
+    // {
+    //   kumi::for_each( [&](auto& m, auto const& v) { m.push_back(v); }
+    //                 , storage, kumi::flatten_all(value)
+    //                 );
+    // }
+
+    //! @brief Removes the last element of the container.
+    //! Calling pop_back on an empty container results in undefined behavior.
+    //! Iterators and references to the last element, as well as the end() iterator, are invalidated.
+    // EVE_FORCEINLINE void pop_back() noexcept
+    // {
+    //   kumi::for_each([&](auto& m) { m.pop_back(); }, storage);
+    // }
+
+    //! @brief Resizes the container to contain count elements.
+    //!
+    //! If the current size is greater than `n`, the container is reduced to its first `n` elements.
+    //! If the current size is less than `n`, additional elements are appended. Those elements are
+    //! either default-inserted or copied if `value` is provided.
+    //! @param n      new size of the container
+    //! @param value  the value to initialize the new elements with
+    void resize(std::size_t n, value_type value = {})
+    {
+      if(n <= capacity())
+      {
+        using alloc_t = std::allocator_traits<Allocator>;
+        kumi::for_each( [&]<typename T>(T* s, auto m)
+                        {
+                          for(std::size_t i=size();i<n;++i) alloc_t::construct(storage, s+i,m);
+                        }
+                      , storage, value
+                      );
+        storage.size_ = n;
+      }
+      else
+      {
+        soa_vector that(n, value);
+        storage_type::piecewise_copy(that.storage, storage, size());
+        swap(that);
```

Comment:
move please.

Reply:
Same answer than before I guess.

Reply:
Same reply. Please use move instead of swap :)

Reply:
operator=(std::move(that));